robot/impl/local_robot.go

```diff
@@ -649,10 +653,22 @@ func (r *localRobot) newResource(
 	return resInfo.DeprecatedRobotConstructor(ctx, r, conf, resLogger)
 }
 
+func (r *localRobot) getTimeout() time.Duration {
+	if newTimeout := os.Getenv("VIAM_RESOURCE_CONFIGURATION_TIMEOUT"); newTimeout != "" {
+		timeOut, err := time.ParseDuration(newTimeout)
```

Comment:
was this tested? what kinds of inputs are allowed?

Reply:
it was tested locally. input should be any string that can be parsed as a duration. for reference, I ran `VIAM_RESOURCE_CONFIGURATION_TIMEOUT=1ns go run web/cmd/server/main.go <args>` and everything instantly timed out.